crates/transaction-emitter-lib/src/transaction_generator/nft_mint_and_transfer.rs

```diff
@@ -31,28 +39,74 @@ impl NFTMint {
             creator_account,
             collection_name,
             token_name,
+            nft_balance: HashMap::new(),
+        }
+    }
+
+    fn add_txns(
+        &mut self,
+        senders: &mut [&mut LocalAccount],
+        receivers: &mut [&mut LocalAccount],
+        transactions_per_account: u64,
+        requests: &mut Vec<SignedTransaction>,
+    ) {
+        for account in senders {
+            for i in 0..transactions_per_account + 1 {
+                requests.push(if i == 0 {
+                    let creator_account = &mut self.creator_account.lock();
+                    let creator_address = creator_account.address();
+                    *self.nft_balance.entry(account.address()).or_insert(0) +=
+                        transactions_per_account;
+                    create_nft_transfer_request(
+                        creator_account,
+                        account,
+                        creator_address,
+                        &self.collection_name,
+                        &self.token_name,
+                        &self.txn_factory,
+                        transactions_per_account,
+                    )
+                } else {
+                    let receiver = receivers.choose_mut(&mut thread_rng()).unwrap();
+                    let amount = if i == transactions_per_account {
+                        let mut balance = 0;
+                        swap(
+                            self.nft_balance.get_mut(&account.address()).unwrap(),
+                            &mut balance,
+                        );
+                        balance
+                    } else {
+                        *self.nft_balance.get_mut(&account.address()).unwrap() -= 1;
+                        1
+                    };
+                    *self.nft_balance.entry(receiver.address()).or_insert(0) += amount;
+                    create_nft_transfer_request(
+                        account,
+                        receiver,
+                        self.creator_account.lock().address(),
+                        &self.collection_name,
+                        &self.token_name,
+                        &self.txn_factory,
+                        amount,
+                    )
```

Comment:
are you sure these txns are not in paralle? ('cause if so they can fail because the txns in `else` depend on that one in `if`)